addons/sale_expense/models/hr_expense.py

```diff
@@ -7,29 +7,31 @@
 class Expense(models.Model):
     _inherit = "hr.expense"
 
-    sale_order_id = fields.Many2one('sale.order', string='Reinvoice Customer', readonly=True, tracking=True,
-        states={'draft': [('readonly', False)], 'reported': [('readonly', False)]},
+    sale_order_id = fields.Many2one('sale.order', compute='_compute_sale_order_id', store=True, copy=True, string='Reinvoice Customer', readonly=False, tracking=True,
+        states={'approved': [('readonly', True)], 'done': [('readonly', True)], 'refused': [('readonly', True)]},
         # NOTE: only confirmed SO can be selected, but this domain in activated throught the name search with the `sale_expense_all_order`
         # context key. So, this domain is not the one applied.
         domain="[('state', '=', 'sale'), ('company_id', '=', company_id)]",
         help="If the product has an expense policy, it will be reinvoiced on this sales order")
```

Comment:
I don't think the sale_order should be copied from one expense to another.

Reply:
I think so.
@tivisse can you confirm the SO shouldn't be copied from one expense to another ? (then it should maybe be specified as copy=False in earlier versions)